internal/resource/utils/equality.go

```diff
@@ -0,0 +1,21 @@
+package utils
+
+import "github.com/hashicorp/consul/proto-public/pbresource"
+
+func EqualType(a *pbresource.Type, b *pbresource.Type) bool {
```

Comment:
Snap! There's now an equivalent method in the `internal/resource` package 😄

Reply:
I switched to using the ones from the resource package and deleted my custom impl.